tycho-bundles/org.eclipse.tycho.p2.maven.repository/src/main/java/org/eclipse/tycho/repository/local/MirroringArtifactProvider.java

```diff
@@ -251,10 +254,34 @@ protected IStatus downloadMostSpecificNeededFormatOfArtifact(IArtifactKey key)
     protected final IStatus downloadCanonicalArtifact(IArtifactKey key)
             throws ProvisionException, ArtifactSinkException {
         // TODO 397355 ignore ProvisionException.ARTIFACT_EXISTS - artifact may have been added by other thread in the meantime
-        IArtifactSink localSink = localArtifactRepository.newAddingArtifactSink(key);
+        IArtifactDescriptor descriptor = localArtifactRepository.createArtifactDescriptor(key);
+        if (descriptor instanceof ArtifactDescriptor) {
+            ArtifactDescriptor localDescriptor = (ArtifactDescriptor) descriptor;
+            IArtifactDescriptor[] descriptors = remoteProviders.getArtifactDescriptors(key);
+            if (descriptors != null && descriptors.length > 0) {
+                IArtifactDescriptor remoteDescriptor = descriptors[0];
+                copyProperty(TychoConstants.PROP_GROUP_ID, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_ARTIFACT_ID, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_VERSION, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_TYPE, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_CLASSIFIER, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_EXTENSION, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_REPOSITORY, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_PGP_KEYS, remoteDescriptor, localDescriptor);
+                copyProperty(TychoConstants.PROP_PGP_SIGNATURES, remoteDescriptor, localDescriptor);
```

Comment:
Couldn't we copy all properties, except the ones we know are used by p2 to identify the path?

Reply:
I now copy all proeprties

Reply:
IIRC, there are some p2 properties that are not so portable (`artifact.something`) but it doesn't seem too important at that stage as I believe they're actually never used in state-of-art p2 repos. So copying all properties seems fine.